test/integration/discovery-consul.bats

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bats
+
+load helpers
+
+# Address on which Consul will listen (random port between 8000 and 9000).
+CONSUL_HOST=127.0.0.1:$(( ( RANDOM % 1000 )  + 8000 ))
+
+# Container name for integration test
+CONSUL_CONTAINER_NAME=swarm_consul
+
+function start_consul() {
+	run docker_host run -h $CONSUL_CONTAINER_NAME -p $CONSUL_HOST:8500 -d progrium/consul -server -bootstrap-expect 1 -data-dir /test
+	[ "$status" -eq 0 ]
+	sleep 20
```

Comment:
Sleep is not good for waiting. May be you should check with `retry`.

Ahh right, doing `docker_host logs` on the consul container and `grep`ing something out of it might work.